### PR TITLE
games-strategy/ja2-stracciatella: version bump to 0.20.0

### PR DESCRIPTION
**Pull request by misanthropos**
for that a library of header files (miniaudio), will be installed via dev-libs/miniaudio.

### CODE REVIEW

**Comment by thesamesam** [comment]
For the ja2 commit, see https://github.com/gentoo/gentoo/pull/28732#issuecomment-1364478851.

File: dev-libs/miniaudio/miniaudio-0.11.11.ebuild [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DESCRIPTION="A single file library for audio playback and capture."
+HOMEPAGE="https://miniaudio.io"
+COMMIT="a0dc1037f99a643ff5fad7272cd3d6461f2d63fa"
+SRC_URI="https://github.com/mackron/miniaudio/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${PN}-${COMMIT}"
```
Comment (thesamesam) [inline dev-libs/miniaudio/miniaudio-0.11.11.ebuild:10]:
```suggestion
S="${WORKDIR}/${PN}-${COMMIT}"

```

File: dev-libs/miniaudio/miniaudio-0.11.11.ebuild [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+KEYWORDS="~amd64"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+src_install() {
+	insinto /usr/include/${PN}/
+	doins -r *
+}
+
+src_test() {
```
Comment (thesamesam) [inline dev-libs/miniaudio/miniaudio-0.11.11.ebuild:22]: src_test should go before src_install (to match phase execution order)

File: dev-libs/miniaudio/miniaudio-0.11.11.ebuild [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+DESCRIPTION="A single file library for audio playback and capture."
+HOMEPAGE="https://miniaudio.io"
+COMMIT="a0dc1037f99a643ff5fad7272cd3d6461f2d63fa"
+SRC_URI="https://github.com/mackron/miniaudio/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${PN}-${COMMIT}"
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="test"
```
Comment (thesamesam) [inline dev-libs/miniaudio/miniaudio-0.11.11.ebuild:14]: IUSE+RESTRICT aren't doing anything as USE=test isn't used anywhere (it's used for conditional building of tests or test deps, not running tests)

File: dev-libs/miniaudio/miniaudio-0.11.11.ebuild [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+RESTRICT="!test? ( test )"
+
+src_install() {
+	insinto /usr/include/${PN}/
+	doins -r *
+}
+
+src_test() {
+	cd "${BUILD_DIR}/test" || die
```
Comment (thesamesam) [inline dev-libs/miniaudio/miniaudio-0.11.11.ebuild:23]: BUILD_DIR isn't defined and this doesn't work for me. Did you definitely run the tests w/ "ebuild ... clean test install" or FEATURES=test?

Comment (misanthropos) [reply dev-libs/miniaudio/miniaudio-0.11.11.ebuild:24]: ah.. no.. I have not tried any tests.
Sorry. I just left it there.

Comment (misanthropos) [reply dev-libs/miniaudio/miniaudio-0.11.11.ebuild:24]: can we leave out those tests?

Comment (thesamesam) [reply dev-libs/miniaudio/miniaudio-0.11.11.ebuild:24]: Yeah it's header only and no build system being used right now so it's ok. Can revisit another time

Comment (misanthropos) [reply dev-libs/miniaudio/miniaudio-0.11.11.ebuild:24]: should I remove test then?

**Comment by Flowdalic** [comment]
All comments of the miniaudio ebuild are marked as resolved, but I don't see any of them resolved. Am I missing something?

**Comment by misanthropos** [comment]
> All comments of the miniaudio ebuild are marked as resolved, but I don't see any of them resolved. Am I missing something?

My mistake. I believed I did change the miniaudio ebuild (I am sure I did). Somehow that push went south. I just did it again. This should be the amended ebuild.

File: games-strategy/ja2-stracciatella/ja2-stracciatella-0.20.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+	>=dev-lang/lua-5.3:*
+	>=dev-libs/miniaudio-0.11.11
+	>=dev-libs/rapidjson-1.1.0
+"
+RDEPEND="
+	${DEPEND}
+	cdinstall? ( games-strategy/ja2-stracciatella-data )
+"
+
+IUSE="$IUSE"
```
Comment (Flowdalic) [inline games-strategy/ja2-stracciatella/ja2-stracciatella-0.20.0.ebuild:160]: This looks strange and should probably be removed.

File: games-strategy/ja2-stracciatella/ja2-stracciatella-0.20.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+		mycmakeargs+=(
+			-DLOCAL_GTEST_LIB=OFF
+			-DLOCAL_LUA_LIB=OFF
+			-DLOCAL_MAGICENUM_LIB=OFF
+			-DLOCAL_MINIAUDIO_LIB=OFF
+			-DLOCAL_RAPIDJSON_LIB=OFF
+			-DLOCAL_SOL_LIB=OFF
+			-DLOCAL_STRING_THEORY_LIB=OFF
+			-DWITH_MAGICENUM=OFF
+			-DWITH_UNITTESTS=$(usex test)
```
Comment (Flowdalic) [inline games-strategy/ja2-stracciatella/ja2-stracciatella-0.20.0.ebuild:178]: This causes emerge to fail with `USE Flag 'test' not in IUSE for games-strategy/ja2-stracciatella-0.20.0`.

**Review by Flowdalic — CHANGES_REQUESTED** [review]

File: games-strategy/ja2-stracciatella/ja2-stracciatella-0.20.0.ebuild (hunk 6 above)
Comment (misanthropos) [reply games-strategy/ja2-stracciatella/ja2-stracciatella-0.20.0.ebuild]: Thank you! And sorry... this test thing is something I really do not know why it has been there in the first place.

**Comment by thesamesam** [comment]
Thanks, made some changes in the following commits (please take a look):

commit f76d6ead4ed5c973823443edbd1e7f3ec32dd94d
Author: Sam James <sam@gentoo.org>
Date:   Sun Jan 8 04:21:36 2023 +0000

    dev-cpp/magic_enum: new package, add 0.8.2

    Needed for games-strategy/ja2-stracciatella in future, possibly. I packaged
    this given I thought it was required but ended up finding it was toggled
    on unnecessarily.

    It's cheap to build and it's likely it'll be needed in future, so..

    Signed-off-by: Sam James <sam@gentoo.org>

commit 01ab9353cd4fe260fb5bbf096494ccdcde6ed800
Author: Sam James <sam@gentoo.org>
Date:   Sun Jan 8 04:12:57 2023 +0000

    dev-libs/miniaudio: EAPI 8, fix LICENSE

    Signed-off-by: Sam James <sam@gentoo.org>

commit 1adc6e82e8dc310810c784400b805edf18ef957b
Author: Sam James <sam@gentoo.org>
Date:   Sun Jan 8 04:08:33 2023 +0000

    games-strategy/ja2-stracciatella: EAPI 8, fix indentation, tests

    Signed-off-by: Sam James <sam@gentoo.org>

**Comment by thesamesam** [comment]
Could you look into upstream allowing libsmacker from system too? It doesn't seem to properly work right now (no option, etc). Thanks!

**Pull request closed without merging**